src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs

```diff
@@ -377,11 +309,14 @@ private void EmitKeepAliveList()
         {
             // See https://llvm.org/docs/LangRef.html#the-llvm-used-global-variable.
             ReadOnlySpan<LLVMValueRef> llvmUsedSymbols = CollectionsMarshal.AsSpan(_keepAliveList);
-            LLVMTypeRef llvmUsedType = LLVMTypeRef.CreateArray(_ptrType, (uint)llvmUsedSymbols.Length);
-            LLVMValueRef llvmUsedGlobal = _module.AddGlobal(llvmUsedType, "llvm.used");
-            llvmUsedGlobal.Linkage = LLVMLinkage.LLVMAppendingLinkage;
-            llvmUsedGlobal.Section = "llvm.metadata";
-            llvmUsedGlobal.Initializer = LLVMValueRef.CreateConstArray(_ptrType, llvmUsedSymbols);
+            if (llvmUsedSymbols.Length != 0)
```

Comment:
Not a correctness fix per se, since we always have something to keep alive, but it turns out a zero-sized `@llvm.used` is illegal.